vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events_database_table.cc

```diff
@@ -0,0 +1,243 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events_database_table.h"
+
+#include <utility>
+
+#include "base/check.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ads/internal/ads_client_helper.h"
+#include "bat/ads/internal/base/database/database_bind_util.h"
+#include "bat/ads/internal/base/database/database_column_util.h"
+#include "bat/ads/internal/base/database/database_transaction_util.h"
+#include "bat/ads/internal/base/logging_util.h"
+#include "bat/ads/internal/features/text_embedding_features.h"
+
+namespace ads {
+namespace database {
+namespace table {
+
+namespace {
+
+constexpr char kTableName[] = "text_embedding_html_events";
+
+int BindParameters(
+    mojom::DBCommandInfo* command,
+    const TextEmbeddingHtmlEventList& text_embedding_html_events) {
+  DCHECK(command);
+
+  int count = 0;
+
+  int index = 0;
+  for (const auto& text_embedding_html_event : text_embedding_html_events) {
+    BindDouble(command, index++,
+               text_embedding_html_event.created_at.ToDoubleT());
+    BindString(command, index++, text_embedding_html_event.version);
+    BindString(command, index++, text_embedding_html_event.locale);
+    BindString(command, index++, text_embedding_html_event.hashed_key);
+    BindString(command, index++, text_embedding_html_event.embedding);
+
+    count++;
+  }
+
+  return count;
+}
+
+TextEmbeddingEventInfo GetFromRecord(mojom::DBRecordInfo* record) {
+  DCHECK(record);
+
+  TextEmbeddingEventInfo text_embedding_html_event;
+
+  text_embedding_html_event.created_at =
+      base::Time::FromDoubleT(ColumnDouble(record, 0));
```

Comment:
We should not stored times using `FromDoubleT`/`ToDoubleT`, see https://source.chromium.org/chromium/chromium/src/+/main:base/time/time.h;l=56 (legacy will be fixed in future to also not use From/ToDoubleT)